reactor-core/src/main/java/reactor/core/Scannable.java

```diff
@@ -210,6 +210,8 @@
 		 */
 		public static final Attr<Stream<Tuple2<String, String>>> TAGS = new Attr<>(null);
 
+		public static final Attr<Boolean> THREAD_MODIFIER = new Attr<>(false);
```

Comment:
@simonbasle I wonder if "asynchronous" (or the inverse - "synchronous") would be a better term?

Reply:
yeah, @aneveu I think the logic should be reversed actually :/
the most important information for now is whether or not the operator is guaranteed to be synchronous. so the attribute would be `SYNCHRONOUS` and it would default to `false`. Make a note in the javadoc that it returns `false` if the operator can switch threads or if we don't know for sure if it stays on the same thread.

Reply:
@bsideup as seen with @simonbasle, I'm going to revert the logic so that by default all operators which are not modifying their executing thread will return `true` for `Attr.SYNCHRONOUS`, in order for asynchronous operators only to be decorated.

Reply:
yeah, sorry this is a lot of additional work but at least if one forgets to explicitly define this attribute, in the case of eg. Sleuth it wouldn't mistakenly be skipped.